common/src/test/java/com/github/dominik48n/party/server/SwitchServerTest.java

```diff
@@ -0,0 +1,214 @@
+        when(partyPlayer.partyId()).thenReturn(Optional.of(partyUuid));
+        when(partyPlayer.uniqueId()).thenReturn(leaderUUID);
+        when(partyProvider.getParty(partyUuid)).thenReturn(Optional.of(party));
+
+        when(switchServerConfig.blackEnable()).thenReturn(true);
+        when(switchServerConfig.blackPatternList()).thenReturn(of(Pattern.compile("^Lobby.*")));
+        when(switchServerConfig.whiteEnable()).thenReturn(true);
+        when(switchServerConfig.whitePatternList()).thenReturn(of(Pattern.compile("Lobby3")));
+
+        switchServer.handleServerConnected(user, serverName);
+
+        verify(partyProvider).connectPartyToServer(party, serverName);
+    }
+
+    @Test
+    void handleServerConnectedFailBecauseNotMatchWhiteList() throws JsonProcessingException {
+        String user = "USER";
+        String serverName = "Lobby2";
+        UUID leaderUUID = UUID.randomUUID();
+        UUID partyUuid = UUID.randomUUID();
+        Party party = createParty(partyUuid, leaderUUID);
+
+        when(userManager.getPlayer(user)).thenReturn(Optional.of(partyPlayer));
+        when(partyPlayer.partyId()).thenReturn(Optional.of(partyUuid));
+        when(partyPlayer.uniqueId()).thenReturn(leaderUUID);
+        when(partyProvider.getParty(partyUuid)).thenReturn(Optional.of(party));
+
+        when(switchServerConfig.blackEnable()).thenReturn(true);
+        when(switchServerConfig.blackPatternList()).thenReturn(of(Pattern.compile("^Lobby.*")));
+        when(switchServerConfig.whiteEnable()).thenReturn(true);
+        when(switchServerConfig.whitePatternList()).thenReturn(of(Pattern.compile("Lobby3")));
+
+        switchServer.handleServerConnected(user, serverName);
+
+        verify(partyProvider, never()).connectPartyToServer(any(), any());
+    }
+
+    @Test
+    void handleServerConnectedDisableWhiteList() throws JsonProcessingException {
+        String user = "USER";
+        String serverName = "Lobby3";
+        UUID leaderUUID = UUID.randomUUID();
+        UUID partyUuid = UUID.randomUUID();
+        Party party = createParty(partyUuid, leaderUUID);
+
+        when(userManager.getPlayer(user)).thenReturn(Optional.of(partyPlayer));
+        when(partyPlayer.partyId()).thenReturn(Optional.of(partyUuid));
+        when(partyPlayer.uniqueId()).thenReturn(leaderUUID);
+        when(partyProvider.getParty(partyUuid)).thenReturn(Optional.of(party));
+
+        when(switchServerConfig.blackEnable()).thenReturn(true);
+        when(switchServerConfig.blackPatternList()).thenReturn(of(Pattern.compile("^Lobby.*")));
+        when(switchServerConfig.whiteEnable()).thenReturn(false);
+
+        switchServer.handleServerConnected(user, serverName);
+
+        verify(partyProvider, never()).connectPartyToServer(any(), any());
+        verify(switchServerConfig, never()).whitePatternList();
+    }
+
+    @Test
+    void handleServerConnectedDisableBlackList() throws JsonProcessingException {
```

Comment:
Please declare the variables in the method that should not be changed as `final`.